Smt/Tactic/Smt.lean

```diff
@@ -74,38 +73,31 @@ def prepareSmtQuery (hints : TSyntax `smtHints) : TacticM Solver := do
   let mut hs ← parseHints hints
   hs := hs.eraseDups
   -- 3. Generate the SMT query.
-  let mut solver := Solver.mk []
-  Query.generateQuery g hs solver
+  Query.generateQuery g hs
 
 @[tactic smt] def evalSmt : Tactic := fun stx => withMainContext do
   let goalType ← Tactic.getMainTarget
-  let solver ← prepareSmtQuery ⟨stx[1]⟩
-  let query := solver.queryToString
+  let query := setOption "produce-models" "true"
+            *> emitCommands (← prepareSmtQuery ⟨stx[1]⟩).reverse
+            *> checkSat
   -- 4. Run the solver.
   let kind := smt.solver.kind.get (← getOptions)
-  let path := smt.solver.path.get? (← getOptions) |>.getD kind.toDefaultPath
-  -- Don't run solver if the server cancelled our task
-  if (← IO.checkCanceled) then return
-  let res ← if let `(smtTimeout| (timeout := $n)) := stx[2] then
```

Comment:
I think this clause has disappeared, let's bring it back.

Reply:
I have a question about this line: Is it enough to check once? (i.e., does Lean have some kind of magic to go back to this line if the server requests cancellation after it has been executed?) If not, it might be better to move this line to `getSexp`...

Reply:
I meant the `stx[2]` line, if the user specifies a timeout we should pass it to `createWithKind`. As for `checkCanceled`, there is no magic - in the current form of cooperative concurrency, we *may* call `checkCanceled` at any point and stop further computation, but it is just a procedure like any other.

Reply:
I see. Then, I will worry about `checkCanceled` later 🙃